### ISSUE TITLE
[cleanup] Replace all instances of `(Cell, F)` with the `AssignedCell` primitive from `halo2`.

### ISSUE THREAD

**Issue by therealyingtong** [appliedzkp/zkevm-circuits#320]
This is safer than passing around tuples, and reduces boilerplate by providing a method for copying assigned cells. An example of where we can use this in the Keccak gadget: https://github.com/appliedzkp/zkevm-circuits/pull/268#discussion_r800561028

**Comment by CPerezz** [comment]
Ohh my bad. I filled #323 since I didn't see this issue. Will close it.

**Comment by silathdiir** [comment]
I'd like to fix this issue if no one is working on it. Thanks.

**Comment by ChihChengLiang** [comment]
@silathdiir Thanks for taking this! This task depends on an upgrade of halo2, so it can be started after the https://github.com/appliedzkp/halo2 side is ready.

**Comment by CPerezz** [comment]
Sorry @silathdiir . I just saw this. 

This required to update halo2 for the entire workspace. As the version we had till now was old in respect to the actual latest release of our fork: https://github.com/appliedzkp/halo2/releases/tag/v2022_02_23

So I took care of the issue and did all the work as required several changes in different crates.

Apologies. #341 is the PR that should solve this. Happy is you want to review it :smiley:
